### PR TITLE
Balance

### PR DESCRIPTION
**Pull request by cmac0351**
These area the components needed for the balance deposit, withdrawal, and transfer.

They can be placed anywhere needed

You need to pass the user id from the db as a prop called 'userId'

<BalanceTransfer userId={userId}/> - this is the Tip Driver page
<BalanceUpdate userId={userId}/> - this is the page where the user can deposit or withdraw funds

### CODE REVIEW

File: client/src/components/Balance/BalanceAPIutils.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import axios from 'axios';
+
+const BalanceAPIutils = {
+  getBalance: (userId) => {
+    return axios.get(`/api/balance/get-balance/${userId}`)
+  },
+
+  deposit: async (userId, amount) => {
+    return axios.put(`/api/balance/deposit/${userId}`, { amount })
+  },
```
Comment (VinhH2402) [inline client/src/components/Balance/BalanceAPIutils.js:10]: have you forgot await on return ?

File: client/src/components/Balance/BalanceTransfer.jsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
+import React, { useState, useEffect } from 'react';
+import CurrentBalance from './CurrentBalance.jsx';
+import BalanceAPIutils from './BalanceAPIutils'
+
+function BalanceTransfer({ userId }) {
```
Comment (sunikkim) [inline client/src/components/Balance/BalanceTransfer.jsx:5]: i wonder what the tradeoffs are between using a functional or class component when there's a lot of state involved? i default to class components when handling a lot of state, but i also haven't practiced using functional components/hooks very much.

**Review by sunikkim — APPROVED** [review]

**Review by Sanoe9 — APPROVED** [review]

File: client/src/components/Balance/BalanceTransfer.jsx [hunk 3]
```diff
@@ -1,14 +1,107 @@
...
+    } catch (err) {
+      if (err.response.status === 405) {
+        setDisplaySelfError(true)
+      }
+      if (err.response.status === 404) {
+        setDisplayNotFoundError(true)
+      }
+    }
+  }
+
```
Comment (VinhH2402) [inline client/src/components/Balance/BalanceTransfer.jsx:45]: I'm not family with useState and useEffect, look like we don't need to do this.setState if we use useState

File: client/src/components/Balance/BalanceTransfer.jsx [hunk 4]
```diff
@@ -1,14 +1,107 @@
...
+          <div className="balance-error">There is no user found with this email address</div>
+          <div className="balance-error">Please Try Again</div>
+        </div>
+      }
+      {displaySuccessMessage &&
+        <div>
+          <div>Success!</div>
+          <div>You just send $ {amount} to {driverEmail}</div>
+          <button type="button" onClick={handleBackToFormClick}>Back to Tip Form</button>
+        </div>
```
Comment (VinhH2402) [inline client/src/components/Balance/BalanceTransfer.jsx:101]: that look very clean with && syntax

**Review by VinhH2402 — APPROVED** [review]
approved

**Pull request merged**